internal/pkg/template/api/service/service.go

```diff
@@ -2,17 +2,55 @@ package service
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/gen/templates"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository/manager"
 )
 
 type Service struct {
-	dependencies dependencies.Container
+	dependencies        dependencies.Container
+	lock                *sync.Mutex
+	repositoriesManager *manager.Manager
 }
 
-func New(d dependencies.Container) templates.Service {
-	return &Service{dependencies: d}
+const TemplateRepositoriesPullInterval = 5 * time.Minute
+
+func New(d dependencies.Container) (*Service, error) {
+	repoManager, err := manager.New(d.Logger())
+	if err != nil {
+		return nil, err
+	}
+	s := &Service{
+		dependencies:        d,
+		lock:                &sync.Mutex{},
+		repositoriesManager: repoManager,
+	}
+	s.StartCron()
+	return s, nil
+}
+
+func (s *Service) StartCron() {
+	go func() {
+		ticker := time.NewTicker(TemplateRepositoriesPullInterval)
```

Comment:
Tu moze byt este problem, s tym, ze to bude bezat vo viacerych instanciach, ktore sa nemusia nastartovat (restartovat) naraz.

Jedna sa nastartuje napr o `9:01` a dalsia `9:05`, ... tj. v jednej moze byt 4 minuty nova verzia, a v druhej stara.
Requesty chodia nahodne, a moze to skoncit na chybe.

V buducnosti budeme musiet asi riesit nejaku synchronizaciu, ...

Ale teraz by stacilo, ak by sa to spustalo kazdych CELYCH 5 minut.

Reply:
Nejak takto (?):

```go
	go func() {
		interval := TemplateRepositoriesPullInterval

		// Delay start to a round time
		startAt := time.Now().Truncate(interval).Add(interval)
		timer := time.NewTimer(time.Until(startAt))
		<-timer.C
		s.dependencies.Logger().Info("pull ticker started")

		// Start ticker
		ticker := time.NewTicker(interval)
		for {
			select {
			case <-s.dependencies.Ctx().Done():
				return
			case <-ticker.C:
				s.pullTemplateRepositories()
			}
		}
	}()
```

Reply:
Supr, funguje 👍